transport_nantes/topicblog/views.py

```diff
@@ -500,3 +500,156 @@ class TopicBlogEmailSend(LoginRequiredMixin, TemplateView):
 
     """
     pass
+
+
+######################################################################
+# TopicBlogPress
+
+class TopicBlogPressViewOnePermissions(PermissionRequiredMixin):
+    """Custom Permission class to require different permissions
+    depending on whether the user is requesting a GET or a POST.
+
+    Default behaviour is at class level and doesn't allow a
+    per-method precision.
+    """
+    def has_permission(self) -> bool:
+        user = self.request.user
+        if self.request.method == 'POST':
+            return user.has_perm('topicblog.tbp.may_publish')
+        elif self.request.method == 'GET':
+            return user.has_perm('topicblog.tbp.may_view')
+        return super().has_permission()
+
+class TopicBlogPressEdit(TopicBlogBaseEdit):
+    model = TopicBlogPress
+    template_name = 'topicblog/tb_item_edit.html'
+    form_class = TopicBlogItemForm
+
+
+class TopicBlogPressView(TopicBlogBaseView):
+    model = TopicBlogPress
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['context_appropriate_base_template'] = 'topicblog/base_press.html'
+        tb_object = context['page']
+        user = self.request.user
+        if user.has_perm('topicblog.tbp.may_send_self') or \
+           (user.has_perm('topicblog.tbp.may_send') \
+            and tb_object.publisher != user):
+            context['sendable'] = True
+        return context
+
+
+class TopicBlogPressViewOne(TopicBlogPressViewOnePermissions,
+                            TopicBlogBaseViewOne):
+    model = TopicBlogPress
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['context_appropriate_base_template'] = 'topicblog/base_press.html'
+        return context
+
+
+class TopicBlogPressList(TopicBlogBaseList):
+    model = TopicBlogPress
+    permission_required = 'topicblog.tbp.may_view'
```

Comment:
There is no PermisionRequiredMixin to enforce this permission. BaseList will ask for Login (LoginRequiredMixin) but no permission is set at this point. So a logged in user would be able to access this view :slightly_smiling_face:

Reply:
It's enough to add `PermisionRequiredMixin` in the inheritance list, right?

Reply:
Yes, no need to have extra permissions checks for this like you did with the [...]One views, imho.

Reply:
Done.